x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/analysis/Analyzer.java

```diff
@@ -750,7 +751,7 @@ private static Expression stringToDate(Expression stringExpression) {
             Long millis = null;
             // TODO: better control over this string format - do we want this to be flexible or always redirect folks to use date parsing
             try {
-                millis = str == null ? null : DateFieldMapper.DEFAULT_DATE_TIME_FORMATTER.parseMillis(str);
+                millis = str == null ? null : convertDatetimeStringToLong(str, ESQL_DEFAULT_DATE_TIME_FORMATTER);
```

Comment:
Using the default methods in EsqlDataTypeConverter this could look like `converterDateTimeStringToLong(str)`.